python/tk_multi_publish2/dialog.py

```diff
@@ -1142,46 +1192,169 @@ def _publish_again_clicked(self):
         # reset the validation flag
         self._validation_run = False
 
-    def _visit_tree_r(self, parent, action, action_name):
+    def _validate_task_generator(self, is_standalone):
```

Comment:
Right now I have 3 separate task generators, one for each of the different passes. The validate has some additional logic because it can be run with/without publish. Otherwise the difference is just the statuses. This feels cleaner and easier to read than trying to do it with a single generator method. Let me know if you disagree.